File: thain-common/src/main/java/com/xiaomi/thain/common/utils/HostUtils.java
package com.xiaomi.thain.common.utils;

import java.net.InetAddress;

/**
 * @author liangyongrui
 */
public class HostUtils {
    private HostUtils() {

    }

    public static String getHostInfo() {
        try {
            return InetAddress.getLocalHost().toString();
        } catch (Exception e) {
            return "unknown";
        }
    }
}

File: thain-common/src/main/kotlin/com/xiaomi/thain/common/utils/HostUtils.kt
package com.xiaomi.thain.common.utils

import java.net.InetAddress

/**
 * @author liangyongrui
 */
object HostUtils {
    val hostInfo: String
        get() = try {
            InetAddress.getLocalHost().toString()
        } catch (e: Exception) {
            "unknown"
        }
}
